### PR TITLE
Export and merge Pontoon terminology for Smartling

### PR DESCRIPTION
**Pull request by bcolsson**

### CODE REVIEW

File: Pontoon/tbx_merge/tbx_merge.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    """Replaces Pontoon IDs with Smartling IDs if the term and definition match exactly a term in Smartling glossary file."""
+    root = etree.getroot()
+    pontoon_term_map = {}
+    for termEntry in root.iter("termEntry"):
+        term = termEntry.find(
+            "./langSet[@{http://www.w3.org/XML/1998/namespace}lang='en-US']/ntig/termGrp/term"
+        ).text
+        try:
+            definition = termEntry.find(
+                "./langSet[@{http://www.w3.org/XML/1998/namespace}lang='en-US']/descripGrp/descrip[@type='definition']"
```
Comment (flodolo) [inline Pontoon/tbx_merge/tbx_merge.py:80]: You can also define a namespace once, and make this a bit more readable

```
nsmap = {"xml": "http://www.w3.org/XML/1998/namespace"}
term = termEntry.find("./langSet[@xml:lang='en-US']/ntig/termGrp/term", nsmap).text
```

File: Pontoon/tbx_merge/tbx_merge.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            ).text
+        except AttributeError:
+            definition = None
+        pontoon_term_map[(term, definition)] = termEntry
```
Comment (flodolo) [inline Pontoon/tbx_merge/tbx_merge.py:84]: Interesting, didn't realize a TBX can have multiple terms with a different definition, and I don't think I've ever used tuples as dictionary keys.

File: Pontoon/tbx_merge/tbx_merge.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        id = termEntry.attrib["id"]
+        smartling_dict[(term, definition)] = id
+
+    return smartling_dict
+
+
+def replace_pontoon_ids(etree, smartling_dict):
+    """Replaces Pontoon IDs with Smartling IDs if the term and definition match exactly a term in Smartling glossary file."""
+    root = etree.getroot()
+    pontoon_term_map = {}
```
Comment (flodolo) [inline Pontoon/tbx_merge/tbx_merge.py:73]: nit: it's a `map` here, but it's a `dict` in `smartling_dict`, should probably be the same.

File: Pontoon/tbx_merge/locales.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+ja
```
Comment (flodolo) [inline Pontoon/tbx_merge/locales.txt:1]: Alphabetical order, please ;-)

File: Pontoon/tbx_merge/tbx_merge.py (hunk 2 above)
Comment (mathjazz) [reply Pontoon/tbx_merge/tbx_merge.py:84]: Yeah, and we actually make use of that TBX feature:
https://pontoon.mozilla.org/it/terminology/common/?list=212229,212230,212243,212244,212245,212246,212292,212293

**Comment by mathjazz** [comment]
Looks good!

Just a quick thought on API and the arguments: If the general use case will be to only generate one export (with either existing, new or smartling IDs), you could replace the `--new` and `--merge-only` arguments with one called something like `--id-strategy`. The possible values would be `keep`, `remove` and `smartling`, the latter one requiring the `--smartling` argument.

**Comment by bcolsson** [comment]
Thanks for the feedback! I've implemented it all here. Didn't seem to break anything so also moved this branch to the pontoon-scripts repo.
https://github.com/bcolsson/scripts/pull/27

**Pull request merged**